### PR TITLE
Fix customer address id not being set on billing address

### PR DESCRIPTION
**Pull request by deefco**
### Description

This patch should fix the issue described in #17705 
### Fixed Issues (if relevant)

1. magento/magento2#17705

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @deefco. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Quote/Model/QuoteManagement.php [hunk 1]
```diff
@@ -598,5 +598,10 @@ protected function _prepareCustomerQuote($quote)
         if ($shipping && !$shipping->getCustomerId() && !$hasDefaultBilling) {
             $shipping->setIsDefaultBilling(true);
         }
+
+        if (!$billing->getCustomerAddressId() && $shippingAddress) {
```
Comment (rogyar) [inline app/code/Magento/Quote/Model/QuoteManagement.php]: The `$shippingAddress` variable might not be set at this point if all previous conditions are not set. We need to check if this variable is set.

File: app/code/Magento/Quote/Model/QuoteManagement.php [hunk 2]
```diff
@@ -598,5 +598,10 @@ protected function _prepareCustomerQuote($quote)
...
+            $billing->setCustomerAddressId($shippingAddress->getId());
+        }
+
```
Comment (rogyar) [inline app/code/Magento/Quote/Model/QuoteManagement.php]: Please, remove this extra line

**Review by rogyar — CHANGES_REQUESTED** [review]

File: app/code/Magento/Quote/Model/QuoteManagement.php (hunk 1 above)
Comment (deefco) [reply app/code/Magento/Quote/Model/QuoteManagement.php]: Changed to $shipping which will either be set to the shipping address or null which means we never enter the check.

File: app/code/Magento/Quote/Model/QuoteManagement.php (hunk 2 above)
Comment (deefco) [reply app/code/Magento/Quote/Model/QuoteManagement.php]: Removed

**Comment by slavvka** [comment]
hey @deefco, thank you fo the contribution! Please fix failed tests. they seem to be failing due to your changes. Also please consider covering the case fixed with a test.

**Comment by slavvka** [comment]
Hey @deefco are you planning to complete this PR?

**Comment by VladimirZaets** [comment]
Hi @deefco, I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Pull request closed without merging**